docs/src/pages/guides/typescript/typescript.md

```diff
@@ -24,265 +24,6 @@ The strict mode options are the same that are required for every types package
-```ts
-const styles = createStyles({
-  root: {
-    minHeight: '100vh',
-  },
-  '@media (min-width: 960px)': {
-    root: {
-      display: 'flex',
-    },
-  },
-});
-```
-
-However to allow these styles to pass TypeScript, the definitions have to be unambiguous concerning names for CSS classes and actual CSS property names. Due to this class names that are equal to CSS properties should be avoided.
-
-```ts
-// error because TypeScript thinks `@media (min-width: 960px)` is a class name
-// and `content` is the CSS property
-const ambiguousStyles = createStyles({
-  content: {
-    minHeight: '100vh',
-  },
-  '@media (min-width: 960px)': {
-    content: {
-      display: 'flex',
-    },
-  },
-});
-
-// works just fine
-const ambiguousStyles = createStyles({
-  contentClass: {
-    minHeight: '100vh',
-  },
-  '@media (min-width: 960px)': {
-    contentClass: {
-      display: 'flex',
-    },
-  },
-});
-```
-
-### Augmenting your props using `WithStyles`
-
-Since a component decorated with `withStyles(styles)` gets a special `classes` prop injected, you will want to define its props accordingly:
-
-```ts
-const styles = (theme: Theme) =>
-  createStyles({
-    root: {
-      /* ... */
-    },
-    paper: {
-      /* ... */
-    },
-    button: {
-      /* ... */
-    },
-  });
-
-interface Props {
-  // non-style props
-  foo: number;
-  bar: boolean;
-  // injected style props
-  classes: {
-    root: string;
-    paper: string;
-    button: string;
-  };
-}
-```
-
-However this isn't very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) because it requires you to maintain the class names (`'root'`, `'paper'`, `'button'`, ...) in two different places. We provide a type operator `WithStyles` to help with this, so that you can just write:
-
-```ts
-import { createStyles } from '@material-ui/styles';
-import { WithStyles } from '@material-ui/core';
-
-const styles = (theme: Theme) =>
-  createStyles({
-    root: {
-      /* ... */
-    },
-    paper: {
-      /* ... */
-    },
-    button: {
-      /* ... */
-    },
-  });
-
-interface Props extends WithStyles<typeof styles> {
-  foo: number;
-  bar: boolean;
-}
-```
-
-### Decorating components
-
-Applying `withStyles(styles)` as a function works as expected:
-
-```tsx
-const DecoratedSFC = withStyles(styles)(({ text, type, color, classes }: Props) => (
-  <Typography variant={type} color={color} classes={classes}>
-    {text}
-  </Typography>
-));
-
-const DecoratedClass = withStyles(styles)(
-  class extends React.Component<Props> {
-    render() {
-      const { text, type, color, classes } = this.props;
-      return (
-        <Typography variant={type} color={color} classes={classes}>
-          {text}
-        </Typography>
-      );
-    }
-  },
-);
-```
-
-Unfortunately due to a [current limitation of TypeScript decorators](https://github.com/Microsoft/TypeScript/issues/4881), `withStyles(styles)` can't be used as a decorator in TypeScript.
-
-## Customization of `Theme`
```

Comment:
@mnajdova Just realized that this section is gone now. Was this accidentally removed or was the move incomplete?

Reply:
See https://github.com/mui-org/material-ui/pull/27417#discussion_r678332315 Almost identical content was already there on the page where I wanted to move it.

Reply:
Let's put this right into the PR description in the future.